src/main/java/nwoolcan/controller/batch/BatchControllerImpl.java

```diff
@@ -36,6 +36,11 @@ public BatchControllerImpl(final Brewery model) {
         this.utils = new ControllerUtils(model);
     }
 
+    @Override
+    public StepController getStepController() {
+        return new StepControllerImpl(this.model);
```

Comment:
Do you really want to create a new controller every time?

Reply:
ok well, you can create it one time i guess